presto-main/src/main/java/io/prestosql/operator/scalar/IpAddressFunctions.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.util.CIDR;
+
+import java.net.UnknownHostException;
+
+import static io.prestosql.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class IpAddressFunctions
+{
+    private IpAddressFunctions() {}
+
+    @Description("Determines whether given IP address exists in the CIDR")
+    @ScalarFunction
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean contains(@SqlType(StandardTypes.VARCHAR) Slice cidrSlice, @SqlType(StandardTypes.IPADDRESS) Slice address)
+    {
+        String cidr = cidrSlice.toStringUtf8();
+        try {
+            return CIDR.contains(cidr, address.getBytes());
+        }
+        catch (IllegalArgumentException | UnknownHostException e) {
```

Comment:
UnknownHostException is (oddly) thrown by InetAddress.getByAddress when "if IP address is of illegal length". So this would happen if `@SqlType(StandardTypes.IPADDRESS) Slice address` is an invalid IP address (anything other than 4 or 16 bytes). In theory, that should never happen, but let's catch it separately and use a different error message for it.